runtime/binding-http-kafka/src/main/java/io/aklivity/zilla/runtime/binding/http/kafka/internal/config/HttpKafkaOptionsConfigAdapter.java

```diff
@@ -134,19 +133,24 @@ public OptionsConfig adaptFromJson(
 
                 if (headers.containsKey(CORRELATION_HEADERS_REPLY_TO_NAME))
                 {
-                    correlationConfig.replyTo(headers.getString(CORRELATION_HEADERS_REPLY_TO_NAME));
+                    replyTo = headers.getString(CORRELATION_HEADERS_REPLY_TO_NAME);
                 }
 
                 if (headers.containsKey(CORRELATION_HEADERS_CORRELATION_ID_NAME))
                 {
-                    correlationConfig.correlationId(headers.getString(CORRELATION_HEADERS_CORRELATION_ID_NAME));
+                    correlationId = headers.getString(CORRELATION_HEADERS_CORRELATION_ID_NAME);
                 }
             }
         }
 
         return HttpKafkaOptionsConfig.builder()
-            .idempotency(idempotencyConfig.build())
-            .correlation(correlationConfig.build())
+            .idempotency()
+                .header(header)
+                .build()
+            .correlation()
+                .replyTo(replyTo)
+                .correlationId(correlationId)
+                .build()
             .build();
```

Comment:
Suggest we create the builder at top of method.

Then, instead of capturing individual temporary variables that might be null, instead call the builder to inject those parts as we discover them in the `if` blocks above.

Then return `builder.build()` here.